pkg/reconciler/eventtype/eventtype.go

```diff
@@ -41,6 +41,11 @@ var _ eventtypereconciler.Interface = (*Reconciler)(nil)
 // 1. Verify the Reference exists.
 func (r *Reconciler) ReconcileKind(ctx context.Context, et *v1beta2.EventType) pkgreconciler.Event {
 
+	// If Reference.Namespace is empty, we will use the Namespace of EventType as the Namespace
+	if et.Spec.Reference.Namespace == "" {
+		et.Spec.Reference.Namespace = et.GetNamespace()
+	}
```

Comment:
This should be done in the webhook (aka in SetDefaults()).

Does it make sense to have event types in one namespace and the reference pointing to another namespace? Should we enforce the use of the same namespace and so validate that the 2 match?

Reply:
I basically have the same/similar assumption as here: 
https://github.com/knative/eventing/blob/main/pkg/reconciler/broker/trigger/trigger.go#L123